### PR TITLE
Add symbolic to Float64 conversion.

### PR DESCRIPTION
**Pull request by torque**
The `N` function had no mechanism to cope with floating point numbers. I've addressed this by aping the way integer conversion is handled.

### CODE REVIEW

**Comment by certik** [comment]
@torque thanks for the PR. Should we add some test for this new functionality?

**Comment by torque** [comment]
Yes, there should be. I will add some tests to this PR shortly.

**Comment by torque** [comment]
I added basic tests for creation, promotion, substitution, and symbolic to numeric conversion. I'm not sure if there are any additional cases that need or should be tested with floating point numbers specifically.

File: src/subs.jl [hunk 1]
```diff
@@ -102,7 +102,7 @@ Convert a SymEngine numeric value into a number
 
 """
 N(b::Basic) = N(BasicType(b))
-N(b::BasicType{Val{:Integer}}) = eval(parse(toString(b)))  ## HACKY
+N(b::BasicNumber) = eval(parse(toString(b)))  ## HACKY
```
Comment (jverzani) [inline src/subs.jl]: The `eval(parse(...))` trick is a way to get integers to parse as a type that depends on the size (so that BigInts) aren't always used. Julia's parser does this, so there may be a more direct way, but I don't know it. However, for floating point numbers this seems to always return a Float64 value. If that is the case, it might be more idiomatic to just call `parse(Float64, string(b))`.  But, in general,  is there a better way to convert a Basic value to a numeric value in SymEngine, then pass that back to Julia.

Comment (isuruf) [inline src/subs.jl]: Yes, For `RealDouble`s we can use `real_double_get_d`. Also there is `basic_eval` that converts Basic values to numeric types in SymEngine. (`RealDouble` and `ComplexDouble`)

**Comment by torque** [comment]
Looking at the comments, I think it would make sense to add ComplexDouble support at the same time. Looking at the SymEngine C API, I don't see an API call that can be used like `real_double_get_d`. I assume complex integer expressions are `eval`'d primarily for the same reason that real integer expressions are. Is there a more appropriate way to do this for doubles?

I also need to add more test cases, as I just realized I didn't modify `walk_expression` so that `lambidfy` would work with the double types.

File: src/numerics.jl [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        else
+            return a
+        end
+    end
+end
+
+N(b::Basic) = N(BasicType(b))
+N(b::BasicType{Val{:RealDouble}}) = convert(Cdouble, b)
+# TODO: Implement rational_get_mpq in cwrapper.h
+N(b::BasicType{Val{:Rational}}) = eval(parse(replace(toString(b), "/", "//")))
```
Comment (isuruf) [inline src/numerics.jl]: This is the only hacky conversion now. We should implement `rational_get_mpq` in `cwrapper.h`

Comment (isuruf) [inline src/numerics.jl]: Opened symengine/symengine#1088 to keep track of this

File: src/numerics.jl [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+##     println("XXX")
+## end
+## need to test for free_symbols, if none then we need to evaluate
+function N(b::BasicType)
+    b = convert(Basic, b)
+    fs = free_symbols(b)
+    if length(fs) > 0
+        throw(ArgumentError("Object can have no free symbols"))
+    end
+    eval(lambdify(b))
```
Comment (jverzani) [inline src/numerics.jl]: I'm not sure either is needed, as in the test cases I tried lambidfy(b) worked as desired.

**Comment by isuruf** [comment]
@torque, thanks for the PR.
